tests/phpunit/tests/block-template-utils.php

```diff
@@ -27,27 +49,170 @@ public static function wpSetUpBeforeClass() {
+		$this->assertEquals( get_stylesheet() . '//my_template_part', $template_part->id );
+		$this->assertEquals( get_stylesheet(), $template_part->theme );
+		$this->assertEquals( 'my_template_part', $template_part->slug );
+		$this->assertEquals( 'publish', $template_part->status );
+		$this->assertEquals( 'custom', $template_part->source );
+		$this->assertEquals( 'My Template Part', $template_part->title );
+		$this->assertEquals( 'Description of my template part', $template_part->description );
+		$this->assertEquals( 'wp_template_part', $template_part->type );
+		$this->assertEquals( WP_TEMPLATE_PART_AREA_HEADER, $template_part->area );
+	}
+
+	function test_inject_theme_attribute_in_block_template_content() {
+		$theme                           = get_stylesheet();
+		$content_without_theme_attribute = '<!-- wp:template-part {"slug":"header","align":"full", "tagName":"header","className":"site-header"} /-->';
+		$template_content                = _inject_theme_attribute_in_block_template_content(
+			$content_without_theme_attribute,
+			$theme
+		);
+		$expected                        = sprintf(
+			'<!-- wp:template-part {"slug":"header","align":"full","tagName":"header","className":"site-header","theme":"%s"} /-->',
+			get_stylesheet()
+		);
+		$this->assertEquals( $expected, $template_content );
+
+		$content_without_theme_attribute_nested = '<!-- wp:group --><!-- wp:template-part {"slug":"header","align":"full", "tagName":"header","className":"site-header"} /--><!-- /wp:group -->';
+		$template_content                       = _inject_theme_attribute_in_block_template_content(
+			$content_without_theme_attribute_nested,
+			$theme
+		);
+		$expected                               = sprintf(
+			'<!-- wp:group --><!-- wp:template-part {"slug":"header","align":"full","tagName":"header","className":"site-header","theme":"%s"} /--><!-- /wp:group -->',
+			get_stylesheet()
+		);
+		$this->assertEquals( $expected, $template_content );
+
+		// Does not inject theme when there is an existing theme attribute.
+		$content_with_existing_theme_attribute = '<!-- wp:template-part {"slug":"header","theme":"fake-theme","align":"full", "tagName":"header","className":"site-header"} /-->';
+		$template_content                      = _inject_theme_attribute_in_block_template_content(
+			$content_with_existing_theme_attribute,
+			$theme
+		);
+		$this->assertEquals( $content_with_existing_theme_attribute, $template_content );
+
+		// Does not inject theme when there is no template part.
+		$content_with_no_template_part = '<!-- wp:post-content /-->';
+		$template_content              = _inject_theme_attribute_in_block_template_content(
+			$content_with_no_template_part,
+			$theme
+		);
+		$this->assertEquals( $content_with_no_template_part, $template_content );
+	}
+
+	/**
+	 * Should retrieve the template from the theme files.
+	 */
+	function test_get_block_template_from_file() {
+		$this->markTestIncomplete();
```

Comment:
Why skipped this? I'm having the same issue in the GB PR.

Reply:
Here It's skipped because there's no FSE theme we can active on Core yet.

Reply:
WP 5.9 will [include Twenty Twenty Two](https://core.trac.wordpress.org/ticket/54318) though 🤔  I guess we should file an issue or ticket somewhere to re-enable these tests once we have that theme available in Core.